apm-agent-plugins/apm-es-restclient-plugin/apm-es-restclient-plugin-common/src/test/java/co/elastic/apm/agent/esrestclient/AbstractEsClientInstrumentationTest.java

```diff
@@ -135,13 +139,13 @@ protected void validateDbContextContent(Span span, List<String> possibleContents
     }
 
     protected void validateSpanContent(Span span, String expectedName, int statusCode, String method) {
-        validateSpanContentWithoutContext(span, expectedName, statusCode, method);
+        validateSpanContentWithoutContext(span, expectedName);
         validateHttpContextContent(span.getContext().getHttp(), statusCode, method);
         validateDestinationContextContent(span.getContext().getDestination());
 
         assertThat(span.getContext().getServiceTarget())
             .hasType(ELASTICSEARCH)
-            .hasNoName()
+            .hasNoName() // we can't validate cluster name here as there is no simple way to inject that without reverse-proxy
```

Comment:
Maybe we can replace the ES testcontainers with ESS deployments?
Not sure how fun it would be to set up and maintain CI secrets to store the credentials though...

Reply:
As this header is injected by cloud proxy, doing a complete cloud deployment would definitely be overkill.
Another rather reasonable approach would be to use a reverse proxy that adds this header in response, but that's still quite a lot for testing a simple HTTP header, this is why I went to testing directly our helper implementation.

Reply:
I didn't think it worth the effort only for this PR, it is a nice alternative to testcontainers due to multiple reasons.
Adding a reverse proxy for testing doesn't satisfy the purpose of testing through ESS, which is to know if this logic ever breaks (eg header name changes). 
Either way, this is certainly out of scope for this PR, only a thought.